src/object/ambient_sound.cpp

```diff
@@ -252,22 +126,94 @@ AmbientSound::collision(GameObject& other, const CollisionHit& hit_)
 void
 AmbientSound::draw(DrawingContext& context)
 {
-  if (Editor::is_active()) {
+  if (Editor::is_active())
     context.color().draw_filled_rect(m_col.m_bbox, Color(0.0f, 0.0f, 1.0f, 0.6f),
                                      0.0f, LAYER_OBJECTS);
-  }
 }
 
 void
 AmbientSound::stop_looping_sounds()
 {
-  stop_playing();
+  if (m_sound_source)
+    m_sound_source->stop(false);
 }
 
 void
 AmbientSound::play_looping_sounds()
 {
-  start_playing();
+  if (Editor::is_active())
+    return;
+
+  m_sound_source->play();
+}
+
+void
+AmbientSound::update(float dt_sec)
+{
+  if (Editor::is_active())
+    return;
```

Comment:
`update()` isn't called in the editor either way. Instead, `editor_update()` is.